### ISSUE TITLE
Classification option needs Logistic Regression (not LassoCV?)

### ISSUE THREAD

**Issue by chriswbartley** [christophM/rulefit#14]
Hi Christoph, Thanks for this code! I'm curious that although it allows Classification based tree generators, when the coefficients are calculated everything goes through LassoCV - but doesn't this use straight SSE (sum of squared error) loss? For binary classification purposes (or OVA multiclass) you'd want L1 regularised Logistic Regression wouldn't you (ie with log loss)? (http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html)? Happy to be corrected if I've not got it right. Friedman's 2005 paper seems a bit vague on this... Cheers, Chris

**Comment by christophM** [comment]
Hi Chris. You are right. Currently it is only implemented for regression task, but not classification. You would want log loss with L1 regularization for classification. 
I'd be happy if you fork the repo and make a PR with some changes !

**Comment by chriswbartley** [comment]
OK, I've done a fairly substantial upgrade to make it more like the R version (http://statweb.stanford.edu/~jhf/r-rulefit/RuleFit_help.html) and to include all the characteristics of the FP2004 paper, I'll put a Pull Request through. I haven't ever done that before, and don't mind if the proposed changes aren't what you are after, I won't be offended :) (it suits my purposes)
